Source/src/Batching/BatchManager.cpp

```diff
@@ -64,7 +71,12 @@ void Hachiko::BatchManager::AddDrawComponent(const ComponentMeshRenderer* mesh_r
 {
     const ResourceMesh* resource = mesh_renderer->GetResourceMesh();
 
-    for (GeometryBatch* geometry_batch : geometry_batches)
+    std::vector<GeometryBatch*>* geometry_batches = 
```

Comment:
wouldn't it be better to use a reference?
std::vector<GeometryBatch*>& geometry_batches =mesh_renderer->GetResourceMaterial()->is_transparent 
                    ? geometry_batches_transparent 
                    : geometry_batches_opaque;

Reply:
@Bernatmago added the required changes, convo is resolved.